cmake/toolchain/zephyr/0.10.0/target.cmake

```diff
@@ -28,5 +29,12 @@ if("${ARCH}" STREQUAL "xtensa")
 
   LIST(APPEND TOOLCHAIN_LIBS hal)
   LIST(APPEND LIB_INCLUDE_DIR -L${SYSROOT_DIR}/lib)
+  set(TOOLCHAIN_HAS_NEWLIB ON CACHE BOOL "True if toolchain supports newlib")
+elseif("${ARCH}" STREQUAL "x86_64")
+  list(APPEND TOOLCHAIN_C_FLAGS -mx32)
+  list(APPEND TOOLCHAIN_LD_FLAGS -mx32)
+  set(TOOLCHAIN_HAS_NEWLIB OFF CACHE BOOL "True if toolchain supports newlib")
```

Comment:
can we put a build of newlib in the SDK for x86_64? AFAIK the network stack currently requires newlib, or at least the tests do

Reply:
newlib doesn't build correctly with `-mx32` on my machine, which results in linking error for `zephyr.elf`. Also, under crosstool-ng, newlib does not have multilib support. This means we could only build for `x86`, `x32` or `x86_64` but not all three. Maybe if we only do `x86` and `x86_64` (without `x32`), newlib might build correctly. We can try again for newlib if we drop `x32`.

Reply:
oof. the kernel is very non 64-bit clean, I think we're going to be using x32 for a while. I agree with @andyross it may be worth investigating what's breaking x32 newlib build, agree its probably something trivial.